src/App.tsx

```diff
@@ -1,34 +1,138 @@
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
-export const App: React.FC = () => {
-  return (
-    <>
-      <div className="section">
-        <div className="container">
-          <div className="box">
-            <h1 className="title">Todos:</h1>
+interface State {
+  todos: Todo[];
+  filteredTodos: Todo[];
+  queryTodos: Todo[];
+  loading: boolean;
+  todoId: number | string;
+  query: string;
+}
 
-            <div className="block">
-              <TodoFilter />
-            </div>
+export class App extends React.Component<{}, State> {
+  state:State = {
+    todos: [],
+    filteredTodos: [],
+    queryTodos: [],
+    loading: true,
+    todoId: 0,
+    query: '',
+  };
+
+  componentDidMount() {
+    getTodos()
+      .then(todos => {
+        this.setState({
+          todos,
+          filteredTodos: todos,
+          queryTodos: todos,
+          loading: false,
+        });
+      });
+  }
+
+  handleChangeAll = () => {
+    this.setState(prevState => {
+      return { filteredTodos: prevState.todos, queryTodos: prevState.todos };
+    });
+  };
+
+  handleChangeCompleted = () => {
+    this.setState(prevState => {
+      const completedTodos
+      = prevState.todos.filter(todo => todo.completed);
+
+      return { filteredTodos: completedTodos, queryTodos: completedTodos };
+    });
+  };
+
+  handleChangeActive = () => {
+    this.setState(prevState => {
+      const active = prevState.todos.filter(todo => !todo.completed);
+
+      return { filteredTodos: active, queryTodos: active };
+    });
+  };
+
+  findQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const query = event.target.value;
+
+    this.setState({ query });
+
+    this.setState(prevState => {
+      const queryTodos
+        = prevState.filteredTodos.filter(todo => todo.title.toLowerCase()
+          .includes(query.toLowerCase().trim()));
+
+      return { queryTodos };
+    });
+  };
+
+  resetQuery = () => {
+    this.setState({ query: '' });
+    this.handleChangeAll();
+  };
+
+  render() {
+    const {
+      queryTodos,
+      loading,
+      todoId,
+      query,
+    } = this.state;
+
+    return (
+      <>
+        <div className="section">
+          <div className="container">
+            <div className="box">
+              <h1 className="title">Todos:</h1>
+
+              <div className="block">
+                <TodoFilter
+                  query={query}
+                  findQuery={this.findQuery}
+                  changeCompleted={this.handleChangeCompleted}
+                  changeActive={this.handleChangeActive}
+                  changeAll={this.handleChangeAll}
+                  resetQuery={this.resetQuery}
+                />
+              </div>
 
-            <div className="block">
-              <Loader />
-              <TodoList />
+              <div className="block">
+                {loading ? <Loader /> : ''}
+                <TodoList
```

Comment:
Let's change a ternary operator a bit. Show the TodoList when the loading is already complete.